pkg/ovn_leader_checker/ovn.go

```diff
@@ -398,3 +422,85 @@ func StartOvnLeaderCheck(cfg *Configuration) {
 		time.Sleep(interval)
 	}
 }
+
+func getTSName(index int) string {
+	if index == 0 {
+		return util.InterconnectionSwitch
+	}
+	return fmt.Sprintf("%s%d", util.InterconnectionSwitch, index)
+}
+
+func getTSCidr(index int) string {
+	cidr := fmt.Sprintf("169.254.%d.0/24", 100+index)
+	return cidr
+}
+
+func updateTS() error {
+	cmdstr := "ovn-ic-nbctl show | wc -l"
+	cmd := exec.Command("sh", "-c", cmdstr)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("ovn-ic-nbctl show output: %s, err: %v", output, err)
+	}
+	lines := strings.Split(string(output), "\n")
+	if len(lines) < 1 {
+		return fmt.Errorf("ovsn-ic-nbctl show count line: err")
+	}
+
+	existTSCount, err := strconv.Atoi(lines[0])
+
+	if err != nil {
+		return fmt.Errorf("existTSCount atoi failed output: %s, err: %v", output, err)
+	}
+
+	expectTSCount, err := strconv.Atoi(os.Getenv("TS_NUM"))
+
+	if err != nil {
+		return fmt.Errorf("expectTSCount atoi failed output: %s, err: %v", output, err)
+	}
+
+	if expectTSCount == existTSCount {
+		return nil
+	}
+
+	if expectTSCount > existTSCount {
+		for i := expectTSCount - 1; i > existTSCount-1; i-- {
+			tsName := getTSName(i)
+			subnet := getTSCidr(i)
+			cmd := exec.Command("ovn-ic-nbctl",
+				ovs.MayExist, "ts-add", tsName,
+				"--", "set", "Transit_Switch", tsName, fmt.Sprintf(`external_ids:subnet="%s"`, subnet))
+			if os.Getenv("ENABLE_SSL") == "true" {
+				cmd = exec.Command("ovn-ic-nbctl",
+					"--private-key=/var/run/tls/key",
+					"--certificate=/var/run/tls/cert",
+					"--ca-cert=/var/run/tls/cacert",
+					ovs.MayExist, "ts-add", tsName,
+					"--", "set", "Transit_Switch", tsName, fmt.Sprintf(`external_ids:subnet="%s"`, subnet))
+			}
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				return fmt.Errorf("output: %s, err: %v", output, err)
+			}
+		}
+	} else if expectTSCount < existTSCount {
```

Comment:
```suggestion
	} else {
```

Reply:
We can also use `func (c LegacyClient) GetTs() ([]string, error)` to get TS count.

Reply:
there will be more code to initalize LegacyClient.